addons/l10n_fr_invoice_addr/views/report_invoice.xml

```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <t t-set="o" t-value="o.with_context(lang=lang)"/>
+
+        <!--
+            ADDRESS BLOCK
+        -->
+
+        <!-- Remove original margin -->
+        <xpath expr="//t[@t-set='address']/address" position="attributes">
+            <attribute name="class">mb-0</attribute>
+        </xpath>
+
+        <!-- SIRET/SIREN -->
+        <xpath expr="(//t[@t-set='address'])[last()]" position="after">
+            <t t-set="address"
+               t-if="o._is_company_french() and o.partner_id.siret">
+                <t t-out="address"/>
+
+                SIRET&#160;: <t t-esc="o.partner_id.siret"/>
+            </t>
+        </xpath>
+
+        <!-- Add final margin -->
+        <xpath expr="(//t[@t-set='address'])[last()]" position="after">
+            <t t-set="address">
+                <t t-out="address"/>
+                <div class="mb-3"/>
+            </t>
+        </xpath>
+
+        <!--
+            INFORMATIONS BLOCK
+        -->
+
+        <!-- Billing Address -->
+        <xpath expr="(//div[@id='informations']/*)[last()]" position="after">
+            <t t-if="o._is_company_french() and o.partner_invoice_id">
+                <t t-set="company_address">
+                    <address t-field="o.partner_id" class="m-0"
+                        t-options='{"widget": "contact", "fields": ["address"], "no_marker": True}'/>
+                </t>
+
+                <t t-set="billing_address">
+                    <address t-field="o.partner_invoice_id" class="m-0"
+                        t-options='{"widget": "contact", "fields": ["address"], "no_marker": True}'/>
+                </t>
+
+                <t t-if="company_address != billing_address">
```

Comment:
i think no need to t-set both if it's just to compare the two, just compare the id directly and then put the <address>